### PR TITLE
More info in RPC errors

### PR DESCRIPTION
**Pull request by buoyad**
before:
screen shot 2017-11-21 at 2 27 51 pm
after:
screen shot 2017-11-21 at 2 27 21 pm

Open to ideas on the error text, I just whipped up what sounded ok with the available information. Broken apart, the contained info is:

1. Code - [this type](https://github.com/keybase/client/blob/master/shared/constants/types/flow-types.js) enumerates what rpc error codes mean
2. Name (e.g. `LOGIN_REQUIRED`) - sometimes(?) passed back by core, corresponds to code
3. Method - the RPC called

r? @keybase/react-hackers

### CODE REVIEW

File: shared/util/errors.js [hunk 1]
```diff
@@ -1,18 +1,38 @@
...
-  code: number
+  code: number // Consult type StatusCode in flow-types.js for what this means
   fields: any
   desc: string // Don't use! This is for compatibility with RPC error object.
+  name: string
 
-  constructor(message: string, code: number, fields: ?any) {
-    super(message || 'Unknown RPC Error')
-    this.code = code
+  constructor(message: string, code: number, fields: ?any, name: ?string, method: ?string) {
```
Comment (akalin-keybase) [inline shared/util/errors.js]: `?any` is just `any`, right?

File: shared/util/errors.js [hunk 2]
```diff
@@ -1,18 +1,38 @@
...
     this.fields = fields
     this.desc = message // Don't use! This is for compatibility with RPC error object.
+    this.name = name || ''
   }
 }
 
+const paramsToErrorMsg: (string, number, ?any, ?string, ?string) => string = (
+  message: string,
+  code: number,
+  fields: ?any,
```
Comment (akalin-keybase) [inline shared/util/errors.js]: here too

File: shared/util/errors.js [hunk 3]
```diff
@@ -1,18 +1,38 @@
...
+  constructor(message: string, code: number, fields: ?any, name: ?string, method: ?string) {
+    super(paramsToErrorMsg(message, code, fields, name, method))
+    this.code = code // Consult type StatusCode in flow-types.js for what this means
     this.fields = fields
     this.desc = message // Don't use! This is for compatibility with RPC error object.
+    this.name = name || ''
   }
 }
 
+const paramsToErrorMsg: (string, number, ?any, ?string, ?string) => string = (
```
Comment (akalin-keybase) [inline shared/util/errors.js]: doesn't it suffice to do

```
const paramsToErrorMsg = (message: string, ...): string => { ... }
```

?

Comment (buoyad) [reply shared/util/errors.js]: Ah yeah, it does. Fixed!

File: shared/util/errors.js (hunk 1 above)
Comment (buoyad) [reply shared/util/errors.js]: Indeed, good catch

**Pull request merged**